server/api/api_accessor.py

```diff
@@ -3,39 +3,47 @@
 import asyncio
 
 import pkg_resources
-from httplib2 import Http
-from oauth2client.service_account import ServiceAccountCredentials
-from server.config import API_TOKEN_URI, API_BASE_URL
+from server.config import API_TOKEN_URI, API_BASE_URL, API_CLIENT_SECRET, API_CLIENT_ID
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+import os
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
 
 CACERTS_FILE = pkg_resources.resource_filename('static', 'cacerts.txt')
 
 
 class ApiAccessor:
-    def __init__(self):
-        self._service_account_credentials = ServiceAccountCredentials.from_p12_keyfile(
-            'faf-server',
-            'faf-server.pem',
-            scopes='write_achievements write_events'
-        )
-        self._service_account_credentials.token_uri = API_TOKEN_URI
-
-    async def api_get(self, path, player_id):
-        loop = asyncio.get_event_loop()
-        result = await loop.run_in_executor(None, partial(self.http(player_id).request, API_BASE_URL + path, "GET"))
-        return result
+    def __init__(self, backend_app_client=BackendApplicationClient, oauth_session=OAuth2Session):
+        client = backend_app_client(client_id=API_CLIENT_ID)
+        self._session = oauth_session(client=client)
+        self._token = self._session.fetch_token(token_url=API_TOKEN_URI, client_id=API_CLIENT_ID, client_secret=API_CLIENT_SECRET)
+
+    async def update_achievements(self, achievements_data, player_id):
```

Comment:
Why `ApiAccessor` has logic (`update_achievements`, `update_events`)?

Reply:
Cause it needs conversion that is related to the API we're communicating with. If the api had been different, the transcription would have been different. If we change the API one day, we wanna have to modify ApiAccessor, not EventService and AchievementService again. I'm trying to make the API-related functions in that class.